crates/bevy_pbr/src/render/morph.rs

```diff
@@ -0,0 +1,92 @@
+use std::{iter, mem};
+
+use bevy_ecs::prelude::*;
+use bevy_render::{
+    mesh::morph::{MorphWeights, MAX_MORPH_WEIGHTS},
+    render_resource::{BufferUsages, BufferVec},
+    renderer::{RenderDevice, RenderQueue},
+    view::ComputedVisibility,
+    Extract,
+};
+use bytemuck::Pod;
+
+#[derive(Component)]
+pub struct Index {
+    pub(super) index: u32,
+}
+#[derive(Resource)]
+pub struct Uniform {
```

Comment:
This is a public type intended to be used outside of this file in a larger context. It should have a less ambiguous name, such as `MorphWeightUniform`. We generally don't use the "require module names to disambiguate types" pattern in Bevy and I'm personally not a huge fan of it. Same goes for Index.

Reply:
Also the `extract` and `prepare` systems. System types should be roughly self-describing, with or without full paths.